### ISSUE TITLE
SQL Lab error handling

### ISSUE THREAD

**Issue by thaume** [apache/incubator-superset#3607]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.20.0

### Expected results
SQL lab show error details when query fails.

### Actual results
Error handling in the SQL Lab view. When an SQL query fails, I just receive a message "could not connect to the server" :
<img width="342" alt="screen shot 2017-10-05 at 7 06 41 pm" src="https://user-images.githubusercontent.com/1096014/31240365-b5bde964-aa00-11e7-919e-4c2a61f629a0.png">

When I check the chrome console, under the XHR network logs, I have a nice error message :
<img width="922" alt="screen shot 2017-10-05 at 7 06 58 pm" src="https://user-images.githubusercontent.com/1096014/31240393-ce9fddde-aa00-11e7-8e44-b5c0bd408b4d.png">

I just want to make sure this is a real problem, then I can start working on a PR.

### Steps to reproduce
Just write a failing query in SQL Lab.

**Comment by xrmx** [comment]
PR welcome, the proper error looks more helpful than a fixed message.

**Comment by mistercrunch** [comment]
My guess is that it's something browser-specific, which browser are you using?

Somehow the error message handling goes wrong somewhere around here in your setup/version.
https://github.com/apache/incubator-superset/blob/master/superset/assets/javascripts/SqlLab/actions.js#L157

**Comment by thaume** [comment]
Thanks for the replies. @mistercrunch I'm using Chrome 61.

First I though that my loadbalancer was eating the error message but it appears not to since I can see the error in the network console.

I'll try to see how I can fix that, will open a PR if I find anything

**Comment by thaume** [comment]
@mistercrunch I'm running some tests, and I'm just wondering if by any chance the error object is not missing the `responseJSON` top object ?

I don't know if it is only for the tests, but this works :
```
// From /superset/assets/spec/javascripts/sqllab/actions_spec.js
it.only('return the correct error message on ajax error', () => {
      ajaxStub.yieldsTo('error', { responseJSON: {
        "query_id": 52,
        "status": "failed",
        "error": "relation \"javascrpt.pages\" does not exist\nLINE 12:    JOIN javascrpt.pages ON pages.anonymous_id = identifies.a...\n                 ^\n"
      }});
      makeRequest();
      expect(dispatch.callCount).to.equal(2);
      console.log("Query failed res: ", dispatch.getCall(1));
      expect(dispatch.getCall(1).args[0].type).to.equal(actions.QUERY_FAILED);
});
```
Error message:
```
args:
   [ { type: 'QUERY_FAILED',
       query: [Object],
       msg: 'relation "javascrpt.pages" does not exist\nLINE 12:    JOIN javascrpt.pages ON pages.anonymous_id = identifies.a...\n                 ^\n' } ],
```

While this fails :
```
// From /superset/assets/spec/javascripts/sqllab/actions_spec.js
it.only('return the correct error message on ajax error', () => {
      ajaxStub.yieldsTo('error', {
        "query_id": 52,
        "status": "failed",
        "error": "relation \"javascrpt.pages\" does not exist\nLINE 12:    JOIN javascrpt.pages ON pages.anonymous_id = identifies.a...\n                 ^\n"
      });
      makeRequest();
      expect(dispatch.callCount).to.equal(2);
      console.log("Query failed res: ", dispatch.getCall(1));
      expect(dispatch.getCall(1).args[0].type).to.equal(actions.QUERY_FAILED);
});
```
Error message :
```
args:
   [ { type: 'QUERY_FAILED',
       query: [Object],
       msg: 'Failed at retrieving results from the results backend' } ],
```

**Comment by t0mdicks0n** [comment]
I am getting the exact same error when querying in SQL Lab in Chrome (version 61.0.3163.100) but not in Safari! Just like you thaume I am able to see the error message in the respone tab in Google Chrome though. My colleagues all experience the same thing.

Maybe an issue with the client for Chrome?

**Comment by xrmx** [comment]
Fixed in #3849
